stream.go

```diff
@@ -136,6 +134,8 @@ func (s *Stream) Close() error {
 	s.closedLocal = true
 	s.clLock.Unlock()
 
+	err := s.mp.sendMsg((s.id<<3)|Close, nil, time.Time{})
+
```

Comment:
Was there a reason to do this in the other order? I see a commit message talking about a "write-close" edge case.